packages/vm/test/api/EIPs/eip-4844-blobs.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { Block } from '@ethereumjs/block'
+import { Blockchain } from '@ethereumjs/blockchain'
+import { Common, Hardfork } from '@ethereumjs/common'
+import { BlobEIP4844Transaction } from '@ethereumjs/tx'
+import {
+  Address,
+  blobsToCommitments,
+  blobsToProofs,
+  bytesToPrefixedHexString,
+  commitmentsToVersionedHashes,
+  getBlobs,
+  initKZG,
+  privateToAddress,
+} from '@ethereumjs/util'
+import * as kzg from 'c-kzg'
+import { bytesToHex, hexToBytes } from 'ethereum-cryptography/utils'
+import * as tape from 'tape'
+
+import genesisJSON = require('../../../../client/test/testdata/geth-genesis/eip4844.json')
+import { VM } from '../../../src/vm'
+import { setBalance } from '../utils'
+
+// Hack to detect if running in browser or not
+const isBrowser = new Function('try {return this===window;}catch(e){ return false;}')
+
+const pk = hexToBytes('20'.repeat(32))
+const sender = bytesToPrefixedHexString(privateToAddress(pk))
+if (isBrowser() === false) {
+  try {
+    initKZG(kzg, __dirname + '/../../../../client/src/trustedSetups/devnet4.txt')
+    // eslint-disable-next-line
+    } catch {}
+
+}
+
+tape('EIP4844 tests', (t) => {
+  t.test('should build a block correctly with blobs', async (st) => {
+    const common = Common.fromGethGenesis(genesisJSON, { chain: 'eip4844' })
+    common.setHardfork(Hardfork.Cancun)
+    const genesisBlock = Block.fromBlockData({ header: { gasLimit: 50000 } }, { common })
+    const blockchain = await Blockchain.create({
+      genesisBlock,
+      common,
+      validateBlocks: false,
+      validateConsensus: false,
+    })
+    const vm = await VM.create({ common, blockchain })
+
+    const address = Address.fromString(sender)
+    await setBalance(vm, address, 14680063125000000000n)
+    const vmCopy = await vm.copy()
+
+    const blockBuilder = await vm.buildBlock({
+      parentBlock: genesisBlock,
+      withdrawals: [],
+      blockOpts: {
+        calcDifficultyFromHeader: genesisBlock.header,
+        freeze: false,
+      },
+      headerData: {
+        gasLimit: 0xffffn,
+      },
+    })
+
+    // Set up tx
+    const blobs = getBlobs('hello world')
+    const commitments = blobsToCommitments(blobs)
+    const versionedHashes = commitmentsToVersionedHashes(commitments)
+    const proofs = blobsToProofs(blobs, commitments)
```

Comment:
Do you think it would be worthwhile creating a generic test utility method for creating blob transactions (like [here]https://github.com/ethereumjs/ethereumjs-monorepo/blob/6d28106ce098e0f445e33dad88f49b208e27d24a/packages/client/test/sim/simutils.ts#L362) but more generic)?  Or maybe we update the static constructor to let you pass in the data for one or more blobs and it generates all of it?  It's just a lot of boilerplate to create these blob transactions (the more I think about it).

Reply:
yes thats a very good idea to have a static method in the tx

Reply:
> yes thats a very good idea to have a static method in the tx OR we can modify fromTxData to just accept blobs string[] and just generate the rest?

Yup, that's more or less what I was thinking.  Let's tackle as a separate PR though.  Not crucial for this one.